controllers/compassmanager_controller.go

```diff
@@ -0,0 +1,213 @@
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=compassmanagers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=compassmanagers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=compassmanagers/finalizers,verbs=update
+//+kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=get;list;watch
+//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=kymas,verbs=get;list;watch;update
+//+kubebuilder:rbac:groups=operator.kyma-project.io,resources=kymas/status,verbs=get
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch
+
+//go:generate mockery --name=Registrator
+type Registrator interface {
+	Register(nameFromKymaCR string) (string, error)
+	ConfigureRuntimeAgent(kubeconfigSecretName string) error
+}
+
+type CompassRegistrator struct{}
+
+func (r *CompassRegistrator) ConfigureRuntimeAgent(kubeconfigSecretName string) error {
+	return nil
+}
+
+func (r *CompassRegistrator) Register(nameFromKymaCR string) (string, error) {
+	return "compass-id", nil
+}
+
+//go:generate mockery --name=Client
+type Client interface {
+	Get(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error
+	Update(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error
+	Create(ctx context.Context, obj client.Object, opts ...client.CreateOption) error
+}
+
+// CompassManagerReconciler reconciles a CompassManager object
+type CompassManagerReconciler struct {
+	Client      Client
+	Scheme      *runtime.Scheme
+	Log         *log.Logger
+	Registrator Registrator
+}
+
+func NewCompassManagerReconciler(mgr manager.Manager, log *log.Logger, r Registrator) *CompassManagerReconciler {
+	return &CompassManagerReconciler{
+		Client:      mgr.GetClient(),
+		Scheme:      mgr.GetScheme(),
+		Log:         log,
+		Registrator: r,
+	}
+}
+
+var ommitStatusChanged = predicate.Or(
+	predicate.GenerationChangedPredicate{},
+	predicate.LabelChangedPredicate{},
+	predicate.AnnotationChangedPredicate{},
+)
+
+var requeueTime = time.Second * 10
+
+func (cm *CompassManagerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// if call to Director will not succeed trigger call once again, and if failed again wait 2 minutes and repeat whole process
+
+	cm.Log.Infof("reconciliation triggered for resource named: %s", req.Name)
+	kymaName := req.Name
+	kubeconfigSecretName := "kubeconfig-" + req.Name
```

Comment:
I don't think it isa good idea to build the secret name. The more bulletproof solution would be to find the secret based on a label.